From 4b5e5c3833461ab2df6842b402193f3efa3ae39d Mon Sep 17 00:00:00 2001
From: Chisoon Jeong <chisoon.jeong@lge.com>
Date: Tue, 12 Sep 2023 08:35:43 +0000
Subject: [PATCH] Backport "Initialize ServerCvc with designated initializers."

This fixes a build error due to a C++20-style parenthesized
initialization of a struct. The patch will be included in upstream's 119
release.

Upstream-Status: Backport [https://crrev.com/c/4855865]
Signed-off-by: Max Ihlenfeldt <max@igalia.com>
---
 .../core/browser/webdata/autofill_sync_bridge_util.cc  | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/components/autofill/core/browser/webdata/autofill_sync_bridge_util.cc b/components/autofill/core/browser/webdata/autofill_sync_bridge_util.cc
index 634443f..f67b901 100644
--- a/components/autofill/core/browser/webdata/autofill_sync_bridge_util.cc
+++ b/components/autofill/core/browser/webdata/autofill_sync_bridge_util.cc
@@ -553,11 +553,13 @@ ServerCvc AutofillWalletCvcStructDataFromWalletCredentialSpecifics(
   base::StringToInt64(wallet_credential_specifics.instrument_id(),
                       &instrument_id);

-  return ServerCvc(
-      instrument_id, base::UTF8ToUTF16(wallet_credential_specifics.cvc()),
-      base::Time::UnixEpoch() +
+  return ServerCvc{
+      .instrument_id = instrument_id,
+      .cvc = base::UTF8ToUTF16(wallet_credential_specifics.cvc()),
+      .last_updated_timestamp =
+          base::Time::UnixEpoch() +
           base::Milliseconds(wallet_credential_specifics
-                                 .last_updated_time_unix_epoch_millis()));
+                                 .last_updated_time_unix_epoch_millis())};
 }

 VirtualCardUsageData VirtualCardUsageDataFromUsageSpecifics(
